Normalize browser e2e

NSeydoux opened the pull request:
This aligns the browser-based end-to-end tests with our other repositories. In particular, this means:
- The tests use a dedicated test app, `e2e/browser/testApp`
- The tests log the user in, and then once the UI changes to indicate the login success they move forward to testing the code actually intended for test.

NSeydoux committed:
Set up new test app

NSeydoux committed:
Align test with other repos

NSeydoux committed:
Fix CI script

NSeydoux committed:
fixup! Fix CI script

NSeydoux committed:
Revert playwright to run headless

NSeydoux committed:
Remove codesandbox

solid-akb commented on e2e/browser/testApp/components/authenticatedFetch/index.tsx:
```diff
@@ -0,0 +1,93 @@
+      getPodUrlAll(session.info.webId, {
+        fetch: session.fetch,
+      })
+        .then((pods) => {
```
Love logging this error here, even though it wouldn't trigger the catch. 👍

solid-akb reviewed (commented):
This looks like a very satisfying PR. Nice!

ThisIsMissEm reviewed (changes requested):
A few changes needed, specifically
- missing dependency (eslint)
- usage of the test function outside of the testing lifecycle (the useEffect)
- commented lines & FIXMEs
- UUIDs
- console.log's

NSeydoux committed:
Use randomUUID from Node core

Replace manually generating a UUID by using a dedicated utility.

NSeydoux committed:
Remove console log

NSeydoux committed:
Uncomment waitFor

Waiting for the response makes this waitFor UI element not necessarily critical, but it's better to keep it in case the UI update is slowed.

ThisIsMissEm commented:
This pull request acts as a follow up: https://github.com/inrupt/solid-client-authn-js/pull/2473/files

NSeydoux committed:
Add error rporting to the UI

NSeydoux commented on e2e/browser/testApp/components/authenticatedFetch/index.tsx:
```diff
@@ -0,0 +1,93 @@
+    }
+  };
+
+  useEffect(() => {
```
Right! That's a leftover from this being a demo app too :). All removed.

NSeydoux committed:
Report errors to UI

NSeydoux commented on playwright.config.ts:
```diff
@@ -47,7 +47,7 @@ const config: PlaywrightTestConfig = {
   timeout: process.env.CI ? 3 * 60_000 : 60_000,
   webServer: {
-    command: "npm run start",
+    command: "cd ./e2e/browser/testApp/ && npm run dev -- -p 3001",
     port: 3001,
```
I have set it explicitly to 3001, because leaving this to default it will try 3000 and move to 3001 if 3000 isn't available, which caused the tests to run against a wrong app instance locally that was still running on 3000, so I'd prefer to be explicit and have the test crash if the port is unavailable.

ThisIsMissEm replied:
Okay, that's fair

NSeydoux replied:
I'll let you imagine my confusion trying to figure out why the authn browser-based tests were spinning up the notifications app.

ThisIsMissEm replied:
yeah, totally!

ThisIsMissEm replied:
maybe we can make `npm run dev` actually set the port number?

NSeydoux replied:
I think it's okay that `npm run dev` is a thin layer on top of `next dev`, and options are passed explicitly

NSeydoux committed:
Add missing eslint dependency

NSeydoux commented on e2e/browser/test/fixtures.ts:
```diff
@@ -24,7 +24,7 @@
 import { test as base } from "@playwright/test";
 
-import crypto from "crypto";
+import nodeCrypto from "crypto";
```
https://github.com/inrupt/solid-client-authn-js/pull/2464/commits/96939080e9dfa1d386818cea77b909b3e01c9863 good to know!

NSeydoux commented on e2e/browser/testApp/package.json:
```diff
@@ -0,0 +1,25 @@
+    "@types/react": "^18.0.17",
+    "@types/react-dom": "^18.0.6",
+    "eslint": "^8.22.0",
+    "eslint-config-next": "^12.2.5",
```
https://github.com/inrupt/solid-client-authn-js/pull/2464/commits/0434b4b046c6e7d49b2ee32fc5183ad2b08bc242

ThisIsMissEm commented on e2e/browser/test/pageModels/AppPage.ts:
```diff
@@ -57,23 +57,29 @@ export class AppPage {
     // response textbox element will be removed from the DOM whilst the response
     // is loaded:
-    await this.fetchResponseText.waitFor({ timeout: this.fetchTimeout });
+    // await this.fetchResponseText.waitFor({ timeout: this.fetchTimeout });
```
okay
